ecl/hqlcpp/hqlsource.cpp

```diff
@@ -3108,8 +3108,18 @@ class DiskReadBuilder : public DiskReadBuilderBase
     DiskReadBuilder(HqlCppTranslator & _translator, IHqlExpression *_tableExpr, IHqlExpression *_nameExpr)
         : DiskReadBuilderBase(_translator, _tableExpr, _nameExpr, (_tableExpr->queryChild(2)->getOperator() != no_pipe))
     {
+        bool usesPluggableFiletype = (_tableExpr->queryChild(2)->getOperator() == no_filetype);
```

Comment:
I think I would approach this as follows:

* Add a parameter forceReadGenerically to SourceBuilder::SourceBuilder and set 
        genericDiskRead = (genericDiskReads && canReadGenerically) || forceReadGenerically;

That means the parameter is set early on, so that anything in the constructor can rely on it.  I also would not worry about checking for hthor - since this is a syntax that the user is explicitly using, and you will get a run time error.

Reply:
Done.

In parent constructor calls, if DiskReadBuilderBase's canReadGenerically is explicitly false then I pass an explicit false to forceReadGenerically.  Otherwise, I compute the arg value.